cert-provider/cert-provider.sh

```diff
@@ -0,0 +1,124 @@
+#!/bin/sh
+
+ACME_BIN="$(realpath ~/.acme.sh/acme.sh)"
+
+# colour output helpers...
+reset=$(tput -T xterm sgr0)
+red=$(tput -T xterm setaf 1)
+green=$(tput -T xterm setaf 2)
+yellow=$(tput -T xterm setaf 3)
+blue=$(tput -T xterm setaf 4)
+
+function logError() {
+    echo "${red}[Error]${reset} $1"
+}
+
+function logWarn() {
+    echo "${yellow}[Warn]${reset} $1"
+}
+
+function logInfo() {
+    echo "${blue}[Info]${reset} $1"
+}
+
+function logSuccess() {
+    echo "${green}[Success]${reset} $1"
+}
+
+# join domains with -d separator
+join_domains()  { echo "$1" | sed -e "s/,/ -d /g" -e "s/^/-d /g"; }
+
+DO_CERTS="true"
+if [ -z "$ACTIVE" ]; then
+    logError "ACTIVE must be set. Value can be anything."
+    unset DO_CERTS
+fi
+
+if [ -z "$DOMAINS" ]; then
+    logError "DOMAINS must be set. Value should be a comma-delimited string of domains."
+    unset DO_CERTS
+else
+    ACME_DOMAINS=$(join_domains "$DOMAINS")
+fi
+
+if [ -z "$EMAIL" ]; then
+    logError "EMAIL must be set. Value should be a valid email address."
+    unset DO_CERTS
+fi
+
+if [ -z "$VALIDATION" ]; then
+    logInfo "VALIDATION not set. Using default: http-01"
+    VALIDATION="http-01"
+else
+    case "$VALIDATION" in
+        "http-01")
+            logInfo "Using validation method: $VALIDATION"
+            ;;
+        "dns-01")
+            logError "The DNS validation method is not supported at this time."
+            unset DO_CERTS
+            ;;
+        *)
+            logError "VALIDATION is invalid. Use a valid value: http-01"
+            unset DO_CERTS
+            ;;
+    esac
+fi
+
+if [ -z "$OUTPUT_PEM" ]; then
+    logError "OUTPUT_PEM must be set. Value should be the path to install your certificate to."
+    unset DO_CERTS
+fi
+
+ACME_OPTS=""
```

Comment:
This should be an array

```suggestion
ACME_OPTS=()
```